src/cmd/sign.js

```diff
@@ -88,6 +90,19 @@ export default function sign(
 
       const manifestId = getManifestId(manifestData);
 
+      if (apiUseAddonSubmissionApi && id && !manifestId) {
+        throw new UsageError(
+          `Cannot set custom ID ${id} - addon submission API requires a ` +
+          'custom id be specified in the manifest'
+        );
+      }
+      if (apiUseAddonSubmissionApi && idFromSourceDir && !manifestId) {
+        throw new UsageError(
+          'Cannot use previously auto-generated extension ID: ' +
+          `${idFromSourceDir} - addon submission API ` +
+          'requires a custom id be specified in the manifest'
+        );
+      }
```

Comment:
Yes, it's an additional requirement we added, but only for the `PUT` requests, because the GUID is specified in the url.  We were thinking of the mv3 requirements though - and how if we were going to add this endpoint then may as well align with the future.